test/jdk/java/io/IO/IO.java

```diff
@@ -174,4 +185,40 @@ public void nullConsole(String method) throws Exception {
         assertEquals(1, output.getExitValue());
         output.shouldContain("Exception in thread \"main\" java.io.IOError");
     }
+
+
+    // adapted from https://junit.org/junit5/docs/current/user-guide/#extensions-lifecycle-callbacks-timing-extension
+
+    public static class TimingExtension implements BeforeTestExecutionCallback,
```

Comment:
In passing, this feels like infrastructure that we could use in several areas as it regularly comes up as to how long specific tests take (the "seconds" line recorded by jtreg in the .jtr file is too coarse grain when there are many tests in the same source file).

Reply:
Looking briefly at this, I think we should be able to provide this as a feature of jtreg's support of `junit` test itself. I've created https://bugs.openjdk.org/browse/CODETOOLS-7903751 to investigate this support in jtreg.

Reply:
This is something that I discussed with Christen Stein about. We've tossed around some ideas, he created CODETOOLS-7903752 so we may have a dup.

Reply:
I've closed mine as the duplicate of the other.